src/controllers/products.controller.js

```diff
@@ -0,0 +1,47 @@
+const Products = require("../models/products.model");
+const asyncHandler = require("../utils/asyncHandler.util");
+
+const getProducts = asyncHandler( async (req, res) => {
+    const products = await Products.find({});
+    res.json(products);
+});
+
+const getProductById = asyncHandler( async (req, res) => {
+    res.json(res.product).end();
+});
```

Comment:
If I am not wrong, `.end()` is not required here as we are sending `.json()` already